### PR TITLE
drivers: watchdog: mcux_wwdt: add driver for NXP LPC 55xxx WWDT

### LINKED ISSUE #22704: Implement watchdog driver for lpcxpresso55s69
**Is your enhancement proposal related to a problem? Please describe.**
The `lpcxpresso55s69` board currently lacks support for watchdog in zephyr.

**Describe the solution you'd like**
Implement a shim driver that adapts the mcux lpc wwdt driver to the zephyr watchdog interface. The existing `wdt_mcux_wdog.c` and `wdt_mcux_wdog32.c` shim drivers for kinetis and i.mx rt can be used for reference. Enable `tests/drivers/watchdog` for this board to ensure the watchdog driver builds and functions correctly.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Nephirus**
Adds shim driver for LPC 55xxx Windowed Watchdog Timer (WWDT).

Also updates the "Supported features" table in LPCXpresso55S69
board documentation page.

Signed-off-by: Jiří Keresteš <jiri@kerestes.cz>

### CODE REVIEW

**Comment by Nephirus** [comment]
Pipeline pass depends on this PR: https://github.com/zephyrproject-rtos/hal_nxp/pull/48

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
> Pipeline pass depends on this PR: [zephyrproject-rtos/hal_nxp#48](https://github.com/zephyrproject-rtos/hal_nxp/pull/48)

Please add a commit to this PR that updates west.yml to points to the hal PR. See [here](https://docs.zephyrproject.org/latest/guides/modules.html#changes-to-existing-module) for more details. This will help fix the buildkite failure.

Thanks for the contribution!

File: boards/arm/lpcxpresso55s69/lpcxpresso55s69_ns.dts [hunk 1]
```diff
@@ -79,3 +79,7 @@
 &hs_lspi {
 	status = "okay";
 };
+
+&wwdt0 {
```
Comment (agansari) [inline boards/arm/lpcxpresso55s69/lpcxpresso55s69_ns.dts:83]: does it work on lpcxpresso55s69_cpu0 ? If yes, please add it to that.

File: drivers/watchdog/wdt_mcux_wwdt.c [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+	WWDT_Deinit(base);
+	data->timeout_valid = false;
+	LOG_DBG("Disabled the watchdog");
+
+	return 0;
+}
+
+#define MSEC_TO_WWDT_TICKS(clock_freq, msec) \
+	((uint32_t)(clock_freq * msec / 1000U / 4))
```
Comment (agansari) [inline drivers/watchdog/wdt_mcux_wwdt.c:66]: Please use the macro MSEC_PER_SEC ore something similar.
Also explain 4 with a comment, i.e. device divides source clock by 4.

File: dts/arm/nxp/nxp_lpc55S6x_common.dtsi [hunk 3]
```diff
@@ -210,6 +211,15 @@
 		status = "okay";
 		label = "RNG";
 	};
+
+	wwdt0: watchdog@c000 {
+		compatible = "nxp,lpc-wwdt";
+		reg = <0xc000 0x1000>;
+		interrupts = <0 0>;
+		status = "disabled";
+		clk-divider = <1>;
```
Comment (agansari) [inline dts/arm/nxp/nxp_lpc55S6x_common.dtsi:220]: Is not the clock divider 4 ? If yes, it should be read int the macro above using DT_PROP()

**Review by agansari — CHANGES_REQUESTED** [review]

File: boards/arm/lpcxpresso55s69/lpcxpresso55s69_ns.dts (hunk 1 above)
Comment (Nephirus) [reply boards/arm/lpcxpresso55s69/lpcxpresso55s69_ns.dts:83]: Thanks for suggestion, I've added relevant bits to lpcxpresso55s69_cpu0.dts and lpcxpresso55s69_cpu0.yaml files.

File: drivers/watchdog/wdt_mcux_wwdt.c (hunk 2 above)
Comment (Nephirus) [reply drivers/watchdog/wdt_mcux_wwdt.c]: Done.

File: dts/arm/nxp/nxp_lpc55S6x_common.dtsi (hunk 3 above)
Comment (Nephirus) [reply dts/arm/nxp/nxp_lpc55S6x_common.dtsi:220]: This clock divider is different from the fixed prescaler in the macro above.

**Comment by henrikbrixandersen** [comment]
Please add "Fixes #22704" to the commit introducing the driver.

**Review by agansari — APPROVED** [review]

**Comment by Nephirus** [comment]
> Please add "Fixes #22704" to the commit introducing the driver.

Done, thanks!

File: west.yml [hunk 4]
```diff
@@ -98,7 +98,7 @@ manifest:
       revision: 1c4fdba512b268033a4cf926bddd323866c3261a
       path: tools/net-tools
     - name: hal_nxp
-      revision: 478f03a71256aeb2890e685ebf5a1dc6832a6b5f
+      revision: pull/48/head
```
Comment (henrikbrixandersen) [inline west.yml:101]: This needs to be updated to use a SHA before merging.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: soc/arm/nxp_lpc/lpc55xxx/soc.c [hunk 5]
```diff
@@ -77,6 +77,11 @@ static ALWAYS_INLINE void clock_init(void)
 	RESET_PeripheralReset(kHSLSPI_RST_SHIFT_RSTn);
 #endif
 
+#if DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(wwdt0), nxp_lpc_wwdt, okay)
```
Comment (galak) [inline soc/arm/nxp_lpc/lpc55xxx/soc.c:80]: why us DT macro here v CONFIG_WDT_MCUX_WWDT?

Comment (Nephirus) [reply soc/arm/nxp_lpc/lpc55xxx/soc.c:80]: I2C and SPI clocks are both set up using DT macros, so I did the same thing here. I'm not sure if DT macros are better than CONFIG_WDT_MCUX_WWDT option, should I change it?

File: west.yml (hunk 4 above)
Comment (MaureenHelm) [reply west.yml:101]: The hal PR was merged, so you can update the SHA now.

Comment (Nephirus) [reply west.yml]: Done.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please remove the intermediate commit `west.yml: hal_nxp: Use PR #48 head to fix pipeline`.

**Comment by Nephirus** [comment]
> Please remove the intermediate commit `west.yml: hal_nxp: Use PR #48 head to fix pipeline`.

Done.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**